satellite/metainfo/loop.go

```diff
@@ -246,10 +246,10 @@ func iterateDatabase(ctx context.Context, db PointerDB, observers []*observerCon
 
 				pathElements := storj.SplitPath(rawPath)
 
-				// we are not storing buckets in pointerDB anymore so
-				// it will be projectID/segmentIndex/bucket_name/encrypted_object_path
 				if len(pathElements) < 4 {
-					return LoopError.New("invalid path %q", rawPath)
+					// We skip this path because it belongs to bucket metadata no an
+					// actual object
+					return nil
```

Comment:
I'm worried that this will also break loop because this will end `func(ctx context.Context, it storage.Iterator) error {`:
```
	err = db.Iterate(ctx, storage.IterateOptions{Recurse: true},
		func(ctx context.Context, it storage.Iterator) error {
			var item storage.ListItem

			// iterate over every segment in metainfo
			for it.Next(ctx, &item) {
```

Reply:
oh rats there's an inner for loop

Reply:
i missed it

Reply:
we should replace `return nil` with `continue`

Reply:
Fixed applied in 
https://github.com/storj/storj/commit/40012e5790bd880b3971c44311d759bc3a29c0ec

Reply:
JT done that already on master

Reply:
good catch @mniewrzal